src/xpcc/architecture/platform/driver/core/cortex/driver.xml

```diff
@@ -38,11 +39,22 @@
 		<!-- stm32f7 -->
 		<template device-family="f7" out="linkerscript.ld">stm32/stm32_idtcm.ld.in</template>
 
-		<template>startup.c.in</template>
-		<template>delay.cpp.in</template>
-		<template>syscalls.cpp.in</template>
+		<!-- dealing with C++ -->
 		<static>cxxabi.cpp</static>
+		<!-- dealing with newlib -->
+		<static>syscalls.c</static>
+		<!-- startup script -->
+		<template>startup.c.in</template>
 		<static>default_handler.sx</static>
+		<!-- everything to do with dynamic memory -->
+		<static>heap_table.c</static>
+		<template>heap_newlib.c.in</template>
```

Comment:
> The problem with python is, that it is imho too powerful for configuration files.

Yes, I would like to agree, but it seems like we have written a very simple package manager that states the dependencies in the device files and the packages in our drivers.
@dergraaf's [library-builder](https://github.com/dergraaf/library-builder) uses Python as well, especially for the module options. So does [Conan](http://docs.conan.io/en/latest/conanfile_py.html#using-options).

There is an argument to be made for more power. For example the linkerscripts should actually be decided by the _type_ of memory available instead of switching on the device name:
```xml
<template device-family="f4" device-name="405|407|415|417|427|429|437|439|469|479" out="linkerscript.ld">stm32/stm32_dccm.ld.in</template>
<template device-family="f4" device-name="401|410|411|446" out="linkerscript.ld">stm32/stm32_ram.ld.in</template>
```
The F401 etc do not have Core Coupled Memory, but I cannot query this in our language, since it is too restricted.
Similarly, not having an `else` in our XML language is a bit annoying in the above example.

I would propose leaving the implementation as it is right now, due to the scope of this PR, and investigate this later.
It would make sense to me to split the `driver/core` into `driver/core`, `driver/libc/newlib` and `driver/libc/avrlibc` or similar.
xpcc is still quite bad at integrating with newlib, and I would like to change that in the future.